runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -337,6 +340,10 @@ public static DataflowRunner fromOptions(PipelineOptions options) {
                   new ReflectiveRootOverrideFactory(StreamingUnboundedRead.class, this)))
           .add(
               PTransformOverride.of(
+                  PTransformMatchers.writeWithRunnerDeterminedSharding(),
```

Comment:
For the test to pass, you need to move this override to occur before the `Read.Bounded` override